src/alejandra_cli/src/cli.rs

```diff
@@ -390,12 +390,16 @@ pub fn main() -> std::io::Result<()> {
 
     let formatted_paths = match matches.values_of("include") {
         Some(include) => {
-            let include = include.collect();
+            let include : Vec<_> = include.collect();
             let exclude = match matches.values_of("exclude") {
                 Some(exclude) => exclude.collect(),
                 None => vec![],
             };
 
+            if include.len() == 1 && exclude.len() == 0 && include[0] == "-" {
+                vec![crate::cli::stdin(quiet)];
```

Comment:
should this be preceded by a `return`?

Without the return (or with the `;` at the end of the line) it looks like this will evaluate stdin, then tui/simple

> For utilities that use operands to represent files to be opened for either reading or writing, the '-' operand should be used only to mean standard input (or standard output when it is clear from context that an output file is being specified).

What I understand from this is not that "`-` should always be allowed to specify `stdin`", but that "in the cases where `-` has a meaning at all, the only meaning allowed should be `stdin`"

Though the fix is not that hacky, if you think it's worth the special case, let's do it. The only thing I would change is making it more clear in the help that only one `-` and no excludes should be used for it to work

Reply:
Something like?
```
diff --git a/src/alejandra_cli/src/cli.rs b/src/alejandra_cli/src/cli.rs
index 10edaf6..e6d556f 100644
--- a/src/alejandra_cli/src/cli.rs
+++ b/src/alejandra_cli/src/cli.rs
@@ -398,6 +398,7 @@ pub fn main() -> std::io::Result<()> {

             if include.len() == 1 && exclude.len() == 0 && include[0] == "-" {
                 vec![crate::cli::stdin(quiet)];
+                return Ok(());
             }

             let paths: Vec<String> = crate::find::nix_files(include, exclude);
```
Didn't seem necessary.